src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs

```diff
@@ -248,6 +222,12 @@ private IServiceCollection BuildHostingServices()
             services.AddSingleton(defaultPlatformServices.Application);
             services.AddSingleton(defaultPlatformServices.Runtime);
 
+            if (!string.IsNullOrEmpty(_options.ServerFactoryAssembly))
+            {
+                // Add the server factory
+                this.UseServer(_options.ServerFactoryAssembly);
```

Comment:
It's a little strange to go back and call one of the extensions here. Revert the changes to `UseServer(string)` and just have a private `ResolveServerAssembly(string, IServiceCollection)`.